Add settings for keeper retries during INSERT (#45788)
ClickHouse/ClickHouse: docs/en/operations/settings/settings.md

└───┘
```

## insert_keeper_max_retries

> The setting sets the maximum number of retries for ClickHouse Keeper (or ZooKeeper) requests during insert into replicated MergeTree. Only Keeper requests which failed due to network error, Keeper session timeout or request timeout considered for retries.

Reviewer: ```suggestion
The setting sets the maximum number of retries for ClickHouse Keeper (or ZooKeeper) requests during insert into replicated MergeTree. Only Keeper requests which failed due to network error, Keeper session timeout, or request timeout are considered for retries.
```

Possible values:

-   Positive integer.
-   0 — Retries are disabled

Default value: 0

Keeper request retries are done after some timeout. The timeout is controlled by the following settings: `insert_keeper_retry_initial_backoff_ms`, `insert_keeper_retry_max_backoff_ms`.

> The first retry is done after `insert_keeper_retry_initial_backoff_ms` timeout. The consequent timeouts will be calculated as following:

Reviewer: ```suggestion
The first retry is done after `insert_keeper_retry_initial_backoff_ms` timeout. The consequent timeouts will be calculated as follows:
```

```
timeout = min(insert_keeper_retry_max_backoff_ms, latest_timeout * 2)
```

For example, if `insert_keeper_retry_initial_backoff_ms=100`, `insert_keeper_retry_max_backoff_ms=10000` and `insert_keeper_max_retries=8` then timeouts will be `100, 200, 400, 800, 1600, 3200, 6400, 10000`.

Apart from fault tolerance, the retries aims to provide better user experience, - they allow to avoid returning a user an error during INSERT execution if Keeper is restarted, for example, due to upgrade.

## insert_keeper_retry_initial_backoff_ms {#insert_keeper_retry_initial_backoff_ms}

> Initial timeout(in milliseconds) to retry failed Keeper request during INSERT query execution

Reviewer: ```suggestion
Initial timeout(in milliseconds) to retry a failed Keeper request during INSERT query execution
```

Possible values:

-   Positive integer.
-   0 — No timeout

Default value: 100

## insert_keeper_retry_max_backoff_ms {#insert_keeper_retry_max_backoff_ms}

> Maximum timeout(in milliseconds) to retry failed Keeper request during INSERT into Rquery execution

Reviewer: ```suggestion
Maximum timeout (in milliseconds) to retry failed Keeper requests during INSERT query execution
```

## background_schedule_pool_size {#background_schedule_pool_size}

> Sets the number of threads performing background tasks for [replicated](../../engines/table-engines/mergetree-family/replication.md) tables, [Kafka](../../engines/table-engines/integrations/kafka.md) streaming, [DNS cache updates](../../operations/server-configuration-parameters/

Author: Not sure why this change is here

Reviewer: I will checkout the PR and see what happened.

Reviewer: LGTM again ;)
